Controller/CouponController.php

```diff
@@ -280,195 +253,209 @@ public function shoppingCoupon(Application $app, Request $request)
                 return $app->redirect($app->url('shopping'));
             }
-
-            if (empty($formCouponCd) && $couponCd) {
-                // 画面上のクーポンコードが入力されておらず、既にクーポンコードが登録されていればクーポンを無効にする
-
-                $this->removeCouponOrder($Order, $app);
-
-                return $app->redirect($app->url('shopping'));
-
-            } else {
+            else {
                 // クーポンコードが入力されている
-
                 $discount = 0;
                 $error = false;
-
+                // クーポン情報を取得
+                /* @var $Coupon Coupon */
+                $Coupon = $app['coupon.repository.coupon']->findActiveCoupon($formCouponCd);
                 if ($app->isGranted('ROLE_USER')) {
                     $Customer = $app->user();
                 } else {
                     $Customer = $app['eccube.service.shopping']->getNonMember($this->sessionKey);
+                    if ($Coupon) {
+                        if ($Coupon->getCouponMember()) {
+                            $form->get('coupon_cd')->addError(new FormError('front.plugin.coupon.shopping.member'));
+                            $error = true;
+                        }
+                    }
                 }
-
-                // クーポン情報を取得
-                $Coupon = $app['eccube.plugin.coupon.repository.coupon']->findActiveCoupon($formCouponCd);
-
-                if ($Coupon) {
-
+                if ($Coupon && !$error) {
+                    $lowerLimit = $Coupon->getCouponLowerLimit();
                     // 既に登録済みのクーポンコードを一旦削除
                     $this->removeCouponOrder($Order, $app);
-
+                    // 対象クーポンが存在しているかチェック
+                    $couponProducts = $service->existsCouponProduct($Coupon, $Order);
+                    $checkLowerLimit = $service->isLowerLimitCoupon($couponProducts, $lowerLimit);
                     // 値引き額を取得
-                    $discount = $service->recalcOrder($Order, $Coupon);
+                    $discount = $service->recalcOrder($Order, $Coupon, $couponProducts);
+                    if (sizeof($couponProducts) == 0) {
+                        $existCoupon = false;
+                    } else {
+                        $existCoupon = true;
+                    }
 
-                    // 対象クーポンが存在しているかチェック
-                    $existCoupon = $service->existsCouponProduct($Coupon, $Order);
                     if (!$existCoupon) {
-                        $form->get("coupon_cd")->addError(new FormError('front.plugin.coupon.shopping.notexists'));
+                        $form->get('coupon_cd')->addError(new FormError('front.plugin.coupon.shopping.notexists'));
+                        $error = true;
+                    }
+
+                    if (!$checkLowerLimit && $existCoupon) {
+                        $message = $app->trans('front.plugin.coupon.shopping.lowerlimit', array('lowerLimit' => number_format($lowerLimit)));
+                        $form->get('coupon_cd')->addError(new FormError($message));
```

Comment:
You could use FormError('front.plugin.coupon.shopping.lowerlimit', null, array('lowerLimit' => number_format($lowerLimit));